loggo/loggo.py

```diff
@@ -377,43 +376,47 @@ def _represent_return_value(self, response: Any, truncate: Optional[int] = 140)
 
         return '({})'.format(self._force_string_and_truncate(response, truncate, use_repr=True))
 
-    def _generate_log(self, where: str, returned: Any, formatters: Dict, safe_log_data: Dict[str, str]) -> None:
+    def _generate_log(self, msg: Optional[str], returned: Any, formatters: Dict, safe_log_data: Dict[str, str]) -> None:
         """
         generate message, level and log data for automated logs
 
-        where (str): 'pre'/'post'/'noreturn'/'error' --- the auto-log type
+        msg (str): the unformatted message
         returned (ANY): what the decorated callable returned
         formatters (dict): dict containing format strings needed for message
         safe_log_data (dict): dict of stringified, truncated, censored parameters
         """
+        # if the user turned off logs of this type, do nothing immediately
+        if not msg:
+            return
+
         # if errors not to be shown and this is an error, quit
-        if not self.allow_errors and where == 'error':
+        if not self.allow_errors and msg == self.errored:
```

Comment:
I think this logic of deciding whether we're logging `'pre'/'post'/'noreturn'/'error'` based on values of `self.errored/self.returned....` etc. is a bit broken. Imagine what happens here if we're logging `'pre'` and the user set the values so that `self.errors == self.called`

Reply:
Yeah, good point, I will have a think about a better way

Reply:
meh, has to pretty much go back the old way, don't like 'special strings' for cases but no good alternative